ee/cis/macos-14/test/scripts/CIS_3.2.sh

```
@@ -1,8 +1,20 @@
-origFlags=$(cat ./tmp.txt  | grep flags: | grep -v naflags);
-sed "s/${origFlags}/flags:-fm,ad,-ex,aa,-fr,lo,-fw/" ./tmp.txt > /etc/security/audit_control;
-rm ./tmp.txt;
+AUDIT_FILE="/etc/security/audit_control"
+TMP_FILE="/tmp/audit_control.tmp"
 
+# Replace existing flags: line, or note its absence for appending
+/usr/bin/sudo /usr/bin/awk '
+    /^flags:/ { print "flags:-fm,ad,-ex,aa,-fr,lo,-fw"; seen=1; next }
+    { print }
+    END { if (!seen) print "flags:-fm,ad,-ex,aa,-fr,lo,-fw" }
+' "$AUDIT_FILE" > "$TMP_FILE"
```

Shellcheck (0.11.0)

[warning] sudo doesn't affect redirects. Use ..| sudo tee file

(SC2024)